src/main/java/org/folio/bulkops/service/BulkOperationService.java

```diff
@@ -721,6 +721,8 @@ private void processDataImportResult(BulkOperation operation) {
       var processedNumOfRecords = metadataProviderService.calculateProgress(executions).getCurrent();
       operation.setProcessedNumOfRecords(operation.getCommittedNumOfErrors() * 2 + processedNumOfRecords);
       if (metadataProviderService.isDataImportJobCompleted(executions)) {
+        operation.setDataImportJobProfileId(null);
+        bulkOperationRepository.save(operation);
```

Comment:
do you really need save here? you have save in switch block

Reply:
It is better to save operation here since saving errors and retrieving identifiers will take some time and may not be finished before next UI poll.

Reply:
errors is sync - client doesn't receive response unless errors are retrieved, but up to you - IMO less changes is better

Reply:
you're right, good catch, fixed.